src/common/lib/python/common_extension.c

```diff
@@ -317,6 +319,20 @@ static int PyTask_init(PyTask *self, PyObject *args, PyObject *kwds) {
     }
   }
   utarray_free(val_repr_ptrs);
+  /* Set the resource vector of the task. */
+  if (resource_vector != NULL) {
+    CHECK(PyList_Size(resource_vector) == MAX_RESOURCE_INDEX);
+    for (int i = 0; i < MAX_RESOURCE_INDEX; ++i) {
+      PyObject *resource_entry = PyList_GetItem(resource_vector, i);
+      task_spec_add_required_resource(self->spec, i,
+                                      PyFloat_AsDouble(resource_entry));
+    }
+  } else {
+    for (int i = 0; i < MAX_RESOURCE_INDEX; ++i) {
+      task_spec_add_required_resource(self->spec, i,
+                                      i == CPU_RESOURCE_INDEX ? 1.0 : 0.0);
```

Comment:
why is it not generalizable? The purpose of this code is to just set the number of requested cores to one and leave everything else zero. This is pretty general, IMHO. It does assume that the default is zero for everything else.